### ISSUE TITLE
Error: Assertion Failed: A helper named 'moment' could not be found

### ISSUE THREAD

**Issue by Jan-Jan** [stefanpenner/ember-moment#50]
When I try to use 

```
{{moment date}}
```

in a hbs file, i get the console error

```
Error: Assertion Failed: A helper named 'moment' could not be found
```

I've installed ember-moment successfully, ie,

```
$ ember install ember-moment
version: 0.2.7
Installed packages for tooling via npm.
installing
Installing browser packages via Bower...
  cached git://github.com/jasonmit/ember-cli-moment-shim.git#0.0.3
  new version for git://github.com/jasonmit/ember-cli-moment-shim.git#*
  resolved git://github.com/jasonmit/ember-cli-moment-shim.git#0.1.0
  cached git://github.com/moment/moment.git#2.10.3
Installed browser packages via Bower.
Installed addon package.
```

Please help

**Comment by artsyca** [comment]
This happens with some addon helpers; They are not registered in integration tests.

The workaround we use is to just register a dummy helper for the tests.

For example:

```coffeescript

`import { test, moduleForComponent } from 'ember-qunit'`
`import { setResolverRegistry } from 'impulse-ember/tests/helpers/set-resolver-registry'`

moduleForComponent 'track-analytics-group/summary', {
  integration: true
  beforeSetup: ->
    setResolverRegistry
      'helper:humanize-number': Ember.HTMLBars.makeBoundHelper((value) -> value)
      'helper:moment'         : Ember.HTMLBars.makeBoundHelper((value) -> value)
}
```

**Comment by artsyca** [comment]
It might not be a great solution but I've been stubbing helpers when necessary:

```coffeescript
  beforeEach: ->
    Ember.Handlebars.helper('moment', ((value) -> value))
    Ember.Handlebars.helper('humanize-number', ((value) -> value))
```

**Comment by jasonmit** [comment]
@artsyca https://github.com/stefanpenner/ember-moment/issues/15 for unit tests.


```
yo git:master ❯ ember install ember-moment                                                                                       ✹ ✭
Future versions of Ember CLI will not support v0.10.28. Please update to Node 0.12 or io.js.
version: 0.2.7
Installed packages for tooling via npm.
installing
Installing browser packages via Bower...
  cached git://github.com/jasonmit/ember-cli-moment-shim.git#0.0.3
  new version for git://github.com/jasonmit/ember-cli-moment-shim.git#*
  resolved git://github.com/jasonmit/ember-cli-moment-shim.git#0.2.0
  cached git://github.com/moment/moment.git#2.9.0
  new version for git://github.com/moment/moment.git#>=2.7.0
  resolved git://github.com/moment/moment.git#2.10.3
Installed browser packages via Bower.
Installed addon package.
```

And the moment helper is working as expected.

Are you sure nothing else is happening within your app which is preventing this initializer from being invoked?  https://github.com/stefanpenner/ember-moment/blob/master/app/initializers/ember-moment.js

**Comment by artsyca** [comment]
@jasonmit not sure. I've noticed several initializers don't work during component integration tests but maybe it's an outdated version of some dependency?

**Comment by jasonmit** [comment]
@stefanpenner are you against me removing the initializer all together and instead depend on the resolver by placing all the helpers in app/helpers/*?

**Comment by awaer** [comment]
I see the same error. After running `ember install ember-moment`, the `moment` template helper is not found. I'm using Ember 1.12.1.

**Comment by jasonmit** [comment]
I haven't been able to reproduce this.  Please show me an example app with the issue so I can look into it.

**Comment by awaer** [comment]
@jasonmit I emailed you an example app.

**Comment by jasonmit** [comment]
@awaer thanks.  Will take a look tonight :+1:

**Comment by jasonmit** [comment]
Found the issue, has to do with the helper not containing a dash.  We previously go around this by manually registering the moment helper on app via an initializer.

```js
var helper = env.helpers[name];
    if (helper) {
      return helper;
    }

    var container = view.container;

    if (!container || ISNT_HELPER_CACHE.get(name)) {
      return; // returns here since `ISNT_HELPER_CACHE` checks for the presence of a dash
    }

    var helperName = "helper:" + name;
    helper = container.lookup(helperName);
    if (!helper) {
      var componentLookup = container.lookup("component-lookup:main");
      Ember['default'].assert("Could not find 'component-lookup:main' on the provided container," + " which is necessary for performing component lookups", componentLookup);

      var Component = componentLookup.lookupFactory(name, container);
      if (Component) {
        helper = makeViewHelper['default'](Component);
        container._registry.register(helperName, helper);
      }
    }
...
```

I'll work on a fix shortly. 1.13.x does not run into this issue

**Comment by jasonmit** [comment]
https://github.com/stefanpenner/ember-moment/pull/64 fixes this, will be versioned as 3.0.1.  Unit testing with anything use the moment helpers will need to either invoke this initializer or manually register the helpers.

The fix has been versioned as 3.0.1
